bin/cron/delete_accounts

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env ruby
+
+require_relative '../../dashboard/config/environment'
+require 'cdo/db'
+require 'cdo/only_one'
+
+OPEN_ENDED_LEVEL_TYPES = %w(
+  Applab
+  FreeResponse
+  Gamelab
+  Weblab
+).freeze
+
+# Deletes all projects and project data associated with a user.
+# @param [Integer] user_id for which to delete projects and project data.
+def delete_projects_and_project_data(user_id)
+  # TODO(asher): Delete channel-backed projects.
+
+  GalleryActivity.where(user_id: user_id).delete_all
+end
+
+# @param [Integer] user_id for which to delete free response answers.
+def delete_open_ended_answers(user_id)
+  UserLevels.where(user_id: user_id).find_each do |user_level|
+    if OPEN_ENDED_LEVEL_TYPES.include? user_level.level.type
+      user_level.level_source.destroy
```

Comment:
It isn't at all clear to me what makes the most sense, from a product or from a engineering perpsective (thus my starting this thread). If we could delete the level source only if there were no user_levels pointing to it, that'd be awesome. Unfortunately, we are lacking the index on `user_levels.level_source_id` to allow us to carry this out efficiently (practically).

Note that we do have an index on `activities.level_source_id`. That said, there would be some complexity in trying to make use of it, as we'd have to consult both `activities` and `activities_old` (the backup table from before our overrunning the ID counter).